### PR TITLE
Cleanup magento/admin-notfication

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

- replaced direct use of object manager
- added missing DocBlocks
- imported FQCN (maybe requires PR for FQCN in API)
- fixed annotations
- added type strict type declaration
- added phpmd/phpcs statements (passes both - MEQP2 not fully)
- changed unit test methods to static
- minor changes in observer, resource model inbox

*Todo*

Fix load/save/delete in loops.

### Fixed Issues (if relevant)

1. n/a

### Manual testing scenarios (*)

1. UnitTests

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19831) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=19831) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: sreichel<br/>:x: Sven Reichel<hr/>**Sven Reichel** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by magento-engcom-team** [comment]
Hi @sreichel. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/CacheOutdatedTest.php [hunk 1]
```diff
@@ -52,26 +68,26 @@ protected function setUp()
...
+        $this->cacheTypeListMock->expects(
+            static::any()
         )->method(
             'getInvalidated'
         )->will(
-            $this->returnValue($cacheTypes)
+            static::returnValue($cacheTypes)
         );
-        $this->assertEquals($expectedSum, $this->_messageModel->getIdentity());
+        static::assertEquals($expectedSum, $this->messageModel->getIdentity());
```
Comment (orlangur) [inline app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/CacheOutdatedTest.php:78]: Such changes to static are invalid and must be reverted.

File: app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/CacheOutdatedTest.php [hunk 2]
```diff
@@ -3,44 +3,60 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+
+declare(strict_types=1);
```
Comment (orlangur) [inline app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/CacheOutdatedTest.php:7]: Please revert such additions, it's better to deliver them separately and do not mix with other changes.

File: app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/CacheOutdatedTest.php [hunk 3]
```diff
@@ -3,44 +3,60 @@
...
+
 namespace Magento\AdminNotification\Test\Unit\Model\System\Message;
 
-class CacheOutdatedTest extends \PHPUnit\Framework\TestCase
+use Magento\AdminNotification\Model\System\Message\CacheOutdated;
+use Magento\Framework\App\Cache\TypeListInterface;
+use Magento\Framework\AuthorizationInterface;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Framework\UrlInterface;
+use PHPUnit\Framework\TestCase;
```
Comment (orlangur) [inline app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/CacheOutdatedTest.php:16]: Please revert FQCN imports, it's better to deliver them separately and do not mix with other changes.

**Review by orlangur — CHANGES_REQUESTED** [review]
Hi @sreichel, please revert all mentioned changes and then rest of changes may be reviewed.

**Pull request closed without merging**